solr/core/src/test/org/apache/solr/schema/TestManagedSchemaWithMultipleAdd.java

```diff
@@ -0,0 +1,106 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.schema;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.CloudSolrClient;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.request.UpdateRequest;
+import org.apache.solr.client.solrj.request.schema.SchemaRequest;
+import org.apache.solr.client.solrj.response.schema.SchemaResponse;
+import org.apache.solr.cloud.SolrCloudTestCase;
+import org.apache.solr.common.SolrInputDocument;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class TestManagedSchemaWithMultipleAdd extends SolrCloudTestCase {
+
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    @BeforeClass
+    public static void createClusterAndInitSysProperties() throws Exception {
+        System.setProperty("managed.schema.mutable", "true");
+        System.setProperty("solr.autoCommit.maxTime", "10000");
+        System.setProperty("solr.autoSoftCommit.maxTime", "3000");
+        configureCluster(1)
+                .addConfig("conf1", TEST_PATH().resolve("configsets").resolve("cloud-managed-autocommit").resolve("conf"))
+                .configure();
+    }
+
+    @AfterClass
+    public static void afterRestartWhileUpdatingTest() {
+        System.clearProperty("managed.schema.mutable");
+        System.clearProperty("solr.autoCommit.maxTime");
+        System.clearProperty("solr.autoSoftCommit.maxTime");
+    }
+
+    @Test
+    public void test() throws Exception {
+        String collection = "testschemaapi";
+        CollectionAdminRequest.createCollection(collection, "conf1", 1, 1)
+                .process(cluster.getSolrClient());
+        testAddFieldAndMultipleDocument(collection);
+    }
+
+    private void testAddFieldAndMultipleDocument(String collection) throws IOException, SolrServerException, InterruptedException {
+
+        CloudSolrClient cloudClient = cluster.getSolrClient();
+
+        String fieldName = "myNewField1";
+        addStringField(fieldName, collection, cloudClient);
+
+        UpdateRequest ureq = new UpdateRequest();
+        int numDocs = 1000;
+        for (int i=0;i<numDocs;i++) {
+            SolrInputDocument doc = new SolrInputDocument();
+            doc.addField("id", i);
+            doc.addField(fieldName, "value" + i);
+            ureq = ureq.add(doc);
+        }
+        cloudClient.request(ureq, collection);
+
+        // Wait for autoCommit to finish if there is one.
```

Comment:
Usually it's better to call an explicit commit in the unit test to avoid timing problems but based on your jira comment manual commit works correctly:

>  If I commit manually (for example with curl) all the documents become available.

You could expand the comment to show that we really need autocommit here.